subsys/net/lib/lwm2m/README_lwm2m

```diff
@@ -0,0 +1,50 @@
+LwM2M Client in Zephyr
+----------------------
+
+### SenML CBOR
+
+SenML CBOR content format support takes advantage of the Zcbor library's capability of generating
+domain specific code. To generate the SenML encoder and decoder the following CDDL file
+is used as an input for the generator:
+
+```
+lwm2m_senml = [1* record]
+
+record = {
+	? bn => tstr,            ; Base Name
+	? n => tstr,             ; Name
+	? ( vi => int .size 8 // ; Integer Value
+	    vf => float       // ; Float Value
+	    vs => tstr        // ; String Value
+	    vb => bool        // ; Boolean Value
+	    vd => bstr ),        ; Data Value
+	0*3 key-value-pair       ; To handle unordered maps; length-first ordered map keys
+}
+
+; now define the generic versions
+key-value-pair = ( int => value )
+
+value = tstr / bstr / int .size 8 / float / bool
+
+n  = 0
+bn = -2
+vi = 2
+vf = 2
+vs = 3
+vd = 8
+vb = 4
+```
+
+The CDDL description is based on the RFC 8428 - Sensor Measurement Lists (SenML). For further
+details please see the chapter 11 - CDDL.
+
+To generate the encoder and decoder:
+
+```console
+zcbor --default-max-qty 99 -c CDDL_FILE code -e -t lwm2m_senml --oc lwm2m_senml_cbor_encode.c --oh lwm2m_senml_cbor_encode.h
+zcbor --default-max-qty 99 -c CDDL_FILE code -d -t lwm2m_senml --oc lwm2m_senml_cbor_decode.c --oh lwm2m_senml_cbor_decode.h
+```
+
+The default value of the number of maximum allowed records doesn't really matter as it's replaced
+with a Kconfig option.
```

Comment:
Please describe any steps taken so they can be correctly done the next time this is regenerated.

Reply:
That's why I put those steps into a separate commit - https://github.com/zephyrproject-rtos/zephyr/pull/44945/commits/1c7e08ef786eda281ac88ce63486e4e53977763e . I'll try to accommodate the same info somehow to this README file.

Reply:
It would be neat if it would be possible to include either a string or an integer when defining max quantity. I could have fed the Kconfig option directly in the code generation phase.

```
#define DEFAULT_MAX_QTY CONFIG_LWM2M_RW_SENML_CBOR_RECORDS
```

Reply:
https://github.com/NordicSemiconductor/zcbor/pull/216

Reply:
High level description given.

Reply:
Sorry, I cannot find the description.